utils/small_vector.hh

```diff
@@ -442,12 +443,15 @@ public:
         std::swap(*this, other);
     }
 
-    bool operator==(const small_vector& other) const noexcept {
-        return size() == other.size() && std::equal(_begin, _end, other.begin());
+    template<typename U = T>
+    requires std::three_way_comparable<U>
+    auto operator<=>(const small_vector& other) const noexcept {
```

Comment:
No need for the fake template. You can add a requires clause between noexcept and `{`.

Reply:
oh, TIL. that's neat! thanks!